gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/ProxyableFolderRetentionPolicy.java

```diff
@@ -0,0 +1,79 @@
+/*
+* Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+* this file except in compliance with the License. You may obtain a copy of the
+* License at  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed
+* under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, either express or implied.
+*/
+
+package gobblin.data.management.retention.policy;
+
+import java.util.Collection;
+import java.util.List;
+
+import azkaban.utils.Props;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+import lombok.AllArgsConstructor;
+
+import org.apache.hadoop.fs.Path;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.data.management.retention.version.StringDatasetVersion;
+import gobblin.util.ProxiedFileSystemUtils;
+
+/**
+ * Implementation of {@link RetentionPolicy} that uses a {@link StringDatasetVersion} and checks if each name of a
+ * {@link StringDatasetVersion} corresponds to a proxyable user.
+ *
+ * <p>
+ *  A proxyable user is defined by the {@link ProxiedFileSystemUtils#canProxyAs(String, String, Path)} method.
+ * <p>
+ */
+@AllArgsConstructor
+public class ProxyableFolderRetentionPolicy implements RetentionPolicy<StringDatasetVersion> {
+
+  private final String userNameToProxyAs;
+  private final String superUserName;
+  private final Path superUserKeytabLocation;
+
+  public ProxyableFolderRetentionPolicy(Props props) {
+    this(props.getString(ConfigurationKeys.FS_PROXY_AS_USER_NAME), props
+        .getString(ConfigurationKeys.SUPER_USER_NAME_TO_PROXY_AS_OTHERS), new Path(
+        props.getString(ConfigurationKeys.SUPER_USER_KEY_TAB_LOCATION)));
+  }
+
+  @Override
+  public Class<? extends DatasetVersion> versionClass() {
+    return StringDatasetVersion.class;
+  }
+
+  @Override
+  public Collection<StringDatasetVersion> listDeletableVersions(List<StringDatasetVersion> allVersions) {
+    return Lists.newArrayList(Iterables.filter(allVersions, new ProxyableVersion()));
+  }
+
+  private class ProxyableVersion implements Predicate<StringDatasetVersion> {
+
+    @Override
+    public boolean apply(StringDatasetVersion input) {
+      return Iterables.all(input.getPathsToDelete(), new ProxyablePath());
+    }
+  }
+
+  private class ProxyablePath implements Predicate<Path> {
+
+    @Override
+    public boolean apply(Path input) {
+      return ProxiedFileSystemUtils.canProxyAs(userNameToProxyAs, superUserName, superUserKeytabLocation);
```

Comment:
Yes, that is true. I think we can use https://hadoop.apache.org/docs/r1.2.1/api/org/apache/hadoop/security/UserGroupInformation.html#getLoginUser() in the `ProxiedFileSystemUtils.loginAndProxyUser` method to prevent the code from re-logging in as the same user. Do you know how to get the equivalent of `[user-name]/[cluster-name]@[realm]` from a UGI object?